### ISSUE TITLE
Hangs if you opt to restart spyder due to resolution change

### ISSUE THREAD

**Issue by rlabbe** [spyder-ide/spyder#10147]
### What steps will reproduce the problem?

1. Alter but do not save a file in the editor

2. Open preferences, change the screen DPI settings (normal to high DPI, or vice versa, depending on your current settings)

3. Press ok. after a few seconds you get a dialog stating that spyder needs to restart to implement the change. Press ok on this dialog.

At this point spyder hangs. I believe this is due to the modified file, as everything works fine if you do not have a modified file open. 

## Versions

* Spyder version: 4.0.0b4 
* Python version: 3.7.3
* Qt version: 5.9.6
* PyQt5 version: 5.9.2
* Operating System: Windows 7

### Dependencies

```
pygments >=2.0    :  2.4.2 (OK)
qdarkstyle >=2.6.4:  2.7 (OK)
sphinx >=0.6.6    :  2.1.2 (OK)
pyls >=0.27.0     :  0.28.2 (OK)
nbconvert >=4.0   :  5.5.0 (OK)
pandas >=0.13.1   :  0.25.1 (OK)
numpy >=1.7       :  1.16.4 (OK)
sympy >=0.7.3     :  1.4 (OK)
cython >=0.21     :  0.29.13 (OK)
qtconsole >=4.5.2 :  4.5.4 (OK)
IPython >=4.0     :  7.8.0 (OK)
matplotlib >=2.0.0:  3.1.1 (OK)
pylint >=0.25     :  2.3.1 (OK)
```

**Comment by ccordoba12** [comment]
Good point. The thing is we show a modal dialog asking users to save modified files before closing Spyder, so an automatic restart can't be completed successfully.

@goanpeca, I think we need to check if this is the case and inform users that an automatic restart can't be performed because they need to save first unsaved work.

**Comment by goanpeca** [comment]
@ccordoba12 I cannot reproduce this. If I follow the described actions I will get a dialog asking if I want to save the unsaved files and clicking yes or no will perform the action and restart as expected.

**Comment by ccordoba12** [comment]
@dalthviz, can you reproduce this on Windows?

**Comment by dalthviz** [comment]
@ccordoba12 @goanpeca I was able reproduce this. To do it I left and unsave untitled file before going to the preferences.
